### ISSUE TITLE
Self referencing Master-Detail relationship

### ISSUE THREAD

**Issue by shakazulu89** [volkanceylan/Serenity#5231]
Hi
I have a situation with an existing database where i need this type of relationship (ill use Orders as a placeholder):
An Order can have "Linked Orders". So it will be a master Order which can have multiple linked orders. I created a "LinkedOrder" table with columns LinkId, MainOrderId, and LinkedOrderId.

The way i want to have it is the user clicks into an order and he will have an editor (like order details) where he can add linked orders. each linked order is essentially a duplicate of the main order but with a different category and different "description". so the user will click to add a linked order, in the popup he will select the order category and write a description.

 **its important to note that the duplicate "linked order" does not yet exist.** 

after the user added the linked orders and clicked update on the main order the linked orders need to be created and the new order IDs need to be updated into the "LinkedOrder" table.

How i went about it is i have a masterdetail relation on OrderRow for LinkedOrder and a simple "LinkedOrder" editor where the user can add linked orders to the main order. then in Order endpoint  before updating/creating i loop the linked orders and create those orders, get the new OrderId and plug it into the LinkedOrders List. 

Im sorry if its hard to follow but i was wondering if this is the correct way to go about it or am i missing some more "built in" that can handle this type of self referencing master/detail situation?

I really appreciate any help/feedback and i am sorry if i am off bases here.
Thanks

**Comment by VictorTomaili** [comment]
Closed because it's does not meet the issue [requirements](https://github.com/volkanceylan/Serenity/wiki/*Issue-Guidelines). 
If you still need help, you can ask this on stackoverflow.
https://stackoverflow.com/tags/serenity-platform
